### PR TITLE
add framework

### PR DESCRIPTION
**Pull request by mgovers**
Adds the basic framework (taken mainly from https://github.com/PowerGridModel/power-grid-model )

* `.gitignore` etc.
* Homogenize licenses
* basic CMake project with header library + C API
* basic python project that loads the C API
* basic CI/github actions

### CODE REVIEW

**Review by Laurynas-Jagutis — DISMISSED** [review]

File: CITATION.cff
Comment (mgovers) [inline CITATION.cff:1]: to be updated in a separate ticket

File: LICENSE
Comment (mgovers) [inline LICENSE:1]: to be reverted

File: README.md [hunk 1]
```diff
@@ -1 +1,27 @@
...
+[![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=PowerGridModel_power-grid-model-io-native&metric=sqale_rating)](https://sonarcloud.io/summary/new_code?id=PowerGridModel_power-grid-model-io-native)
+[![Reliability Rating](https://sonarcloud.io/api/project_badges/measure?project=PowerGridModel_power-grid-model-io-native&metric=reliability_rating)](https://sonarcloud.io/summary/new_code?id=PowerGridModel_power-grid-model-io-native)
+[![Security Rating](https://sonarcloud.io/api/project_badges/measure?project=PowerGridModel_power-grid-model-io-native&metric=security_rating)](https://sonarcloud.io/summary/new_code?id=PowerGridModel_power-grid-model-io-native)
+[![Vulnerabilities](https://sonarcloud.io/api/project_badges/measure?project=PowerGridModel_power-grid-model-io-native&metric=vulnerabilities)](https://sonarcloud.io/summary/new_code?id=PowerGridModel_power-grid-model-io-native)
+
+[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.8054429.svg)](https://zenodo.org/record/8054429)
```
Comment (mgovers) [inline README.md:23]: to be updated

File: cmake/README.md.in [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<!--
+SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+
+SPDX-License-Identifier: MPL-2.0
+-->
+
+# Power Grid Model IO Native
```
Comment (mgovers) [inline cmake/README.md.in:7]: to be updated

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+#     "breathe",
+#     "myst_nb",
+#     "sphinx_rtd_theme",
+#     "readthedocs-sphinx-search",
+#     "sphinx-hoverxref",
+#     "sphinxcontrib-mermaid",
+#     "numpydoc",
+#     "pandas",
+#     "gitpython",
+# ]
```
Comment (mgovers) [inline pyproject.toml:61]: to be added

File: README.md (hunk 1 above)
Comment (mgovers) [reply README.md:23]: Moved to #2

File: CITATION.cff
Comment (mgovers) [reply CITATION.cff:1]: Moved to #2

File: cmake/README.md.in (hunk 2 above)
Comment (mgovers) [reply cmake/README.md.in:7]: Moved to #3

File: pyproject.toml (hunk 3 above)
Comment (mgovers) [reply pyproject.toml:61]: Moved to #3

**Review by Laurynas-Jagutis — APPROVED** [review]

**Pull request merged**